src/CommunityToolkit.Maui.Core/IArgumentsReceiver.cs

```diff
@@ -0,0 +1,13 @@
+﻿namespace CommunityToolkit.Maui.Core;
+
+/// <summary>
+/// Represents an implementation that can receive arguments during specific lifecycle events.
+/// </summary>
+public interface IArgumentsReceiver
+{
+	/// <summary>
+	/// Sets the arguments ready for use.
+	/// </summary>
+	/// <param name="arguments">A set of arguments.</param>
+	void SetArguments(IDictionary<string, object> arguments);
+}
```

Comment:
This might not stick around but the intention is to not rely on `IQueryAttributable` based on an initial discussion with @PureWeen and this open issue/discussion https://github.com/dotnet/maui/issues/10294. I suspect we might opt to move towards something like `IViewAppearing` or similar but I wanted to make a clean break from `IQueryAttributable` for now so we don't misdirect anyone...

Reply:
We could use a [`IReadOnlyDictionary`](https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.ireadonlydictionary-2?view=net-8.0) here, this will ensure the collection is immutable. This is based on feedback that I got for `IQueryAttributable`.

Reply:
Excellent idea!